lib/Runtime/Library/BoundFunction.cpp

```diff
@@ -84,6 +82,12 @@ namespace Js
             // If no "this" is passed, "undefined" is used
             boundThis = scriptContext->GetLibrary()->GetUndefined();
         }
+
+        // Reduce length number of bound args
+        len = len - this->count;
+        len = max(len, 0);
```

Comment:
Could you add a test for this case (where the target function is over-bound)?

Reply:
Done. Though i’ve also realised that i’m missing tests for anonymous functions which take a slightly different path. I’ll add those later and also try and refactor the testfile to  remove some of the  duplication.